test/APIJSON/spi.swift

```diff
@@ -132,200 +134,225 @@ public func spiAvailableFunc() {}
 // CHECK-SPI-NEXT:     {
 // CHECK-SPI-NEXT:       "name": "_$s8MyModule0A6Class2CMn",
 // CHECK-SPI-NEXT:       "access": "public",
-// CHECK-SPI-NEXT:       "file": "/@input/MyModule.swiftmodule",
+// CHECK-SPI-EXTRACT-NEXT:  "file": "/@input/MyModule.swiftmodule",
+// CHECK-SPI-EMIT-NEXT:     "file": "SOURCE_DIR/test/APIJSON/spi.swift",
 // CHECK-SPI-NEXT:       "linkage": "exported"
 // CHECK-SPI-NEXT:     },
 // CHECK-SPI-NEXT:     {
 // CHECK-SPI-NEXT:       "name": "_$s8MyModule0A6Class2CMo",
 // CHECK-SPI-NEXT:       "access": "public",
-// CHECK-SPI-NEXT:       "file": "/@input/MyModule.swiftmodule",
+// CHECK-SPI-EXTRACT-NEXT:  "file": "/@input/MyModule.swiftmodule",
+// CHECK-SPI-EMIT-NEXT:     "file": "SOURCE_DIR/test/APIJSON/spi.swift",
 // CHECK-SPI-NEXT:       "linkage": "exported"
 // CHECK-SPI-NEXT:     },
 // CHECK-SPI-NEXT:     {
 // CHECK-SPI-NEXT:       "name": "_$s8MyModule0A6Class2CMu",
 // CHECK-SPI-NEXT:       "access": "public",
-// CHECK-SPI-NEXT:       "file": "/@input/MyModule.swiftmodule",
+// CHECK-SPI-EXTRACT-NEXT:  "file": "/@input/MyModule.swiftmodule",
+// CHECK-SPI-EMIT-NEXT:     "file": "SOURCE_DIR/test/APIJSON/spi.swift",
 // CHECK-SPI-NEXT:       "linkage": "exported"
 // CHECK-SPI-NEXT:     },
 // CHECK-SPI-NEXT:     {
 // CHECK-SPI-NEXT:       "name": "_$s8MyModule0A6Class2CN",
 // CHECK-SPI-NEXT:       "access": "public",
-// CHECK-SPI-NEXT:       "file": "/@input/MyModule.swiftmodule",
+// CHECK-SPI-EXTRACT-NEXT:  "file": "/@input/MyModule.swiftmodule",
+// CHECK-SPI-EMIT-NEXT:     "file": "SOURCE_DIR/test/APIJSON/spi.swift",
 // CHECK-SPI-NEXT:       "linkage": "exported"
 // CHECK-SPI-NEXT:     },
 // CHECK-SPI-NEXT:     {
 // CHECK-SPI-NEXT:       "name": "_$s8MyModule0A6Class2CfD",
 // CHECK-SPI-NEXT:       "access": "public",
-// CHECK-SPI-NEXT:       "file": "/@input/MyModule.swiftmodule",
+// CHECK-SPI-EXTRACT-NEXT:  "file": "/@input/MyModule.swiftmodule",
+// CHECK-SPI-EMIT-NEXT:     "file": "SOURCE_DIR/test/APIJSON/spi.swift",
 // CHECK-SPI-NEXT:       "linkage": "exported"
 // CHECK-SPI-NEXT:     },
 // CHECK-SPI-NEXT:     {
 // CHECK-SPI-NEXT:       "name": "_$s8MyModule15newUnprovenFuncyyF",
 // CHECK-SPI-NEXT:       "access": "private",
-// CHECK-SPI-NEXT:       "file": "/@input/MyModule.swiftmodule",
+// CHECK-SPI-EXTRACT-NEXT:  "file": "/@input/MyModule.swiftmodule",
+// CHECK-SPI-EMIT-NEXT:     "file": "SOURCE_DIR/test/APIJSON/spi.swift",
 // CHECK-SPI-NEXT:       "linkage": "exported"
 // CHECK-SPI-NEXT:     },
 // CHECK-SPI-NEXT:    {
 // CHECK-SPI-NEXT:       "name": "_$s8MyModule16spiAvailableFuncyyF",
 // CHECK-SPI-NEXT:       "access": "private",
-// CHECK-SPI-NEXT:       "file": "/@input/MyModule.swiftmodule",
+// CHECK-SPI-EXTRACT-NEXT:  "file": "/@input/MyModule.swiftmodule",
+// CHECK-SPI-EMIT-NEXT:     "file": "SOURCE_DIR/test/APIJSON/spi.swift",
 // CHECK-SPI-NEXT:       "linkage": "exported",
 // CHECK-SPI-NEXT:       "introduced": "10.10"
-// CHECK-SPI-NEXT:     },
```

Comment:
It is concerning when a public symbol disappears. Does the symbol `_main` exist in the binary file?

Reply:
`_main` was erroneously added before because the `swift-frontend` invocation did not specify `-parse-as-library` (normally a library would not have a `_main` entry point).